test/support/base_test_model.rb

```diff
@@ -94,7 +94,7 @@ def self.create_default!(*source_associations)
 
   def create_has_one!(association_name, attrs = {})
     association_name = ActiveRecordWhereAssoc::ActiveRecordCompat.normalize_association_name(association_name)
-    reflection = self.class.reflections[association_name]
+    reflection = self.class._reflections[association_name]
```

Comment:
Looks like Rails v7.2 has symbol keys for `._reflections` and string keys for `.reflections`.